src/Catalog.Application/Services/CategoryService.cs

```diff
@@ -41,7 +47,27 @@ public async Task Update(Category category)
     {
         await categoryValidator.ValidateAndThrowAsync(category);
 
-        await categoryRepository.Update(category);
+        await categoryRepository.Update(category);
+
+        await PublishUpdateMessage(category);
+    }
+
+    private async Task PublishUpdateMessage(Category category)
+    {
+        try
+        {
+            await messageSenderService.SendAsync(new Domain.Messages.CategoryUpdatedMessage
+            {
+                Id = category.Id,
+                Name = category.Name,
+                ImageUrl = category.ImageUrl,
+                ParentCategoryId = category.Parent?.Id
+            });
+        }
+        catch (MessageSendoutFailedException ex)
+        {
+            logger.Log(LogLevel.Error, ex, ex?.Message);

```

Comment:
Out of pure curiosity why not use `logger.LogError(ex....)`

Reply:
We discussed this on the session that it was not intentional and the logger.LogError method can be used as well